packages/cli/src/telemetry.ts

```diff
@@ -0,0 +1,209 @@
+import { networkInterfaces } from 'os';
+import { randomBytes, createHash, BinaryLike } from 'crypto';
+import * as os from 'os';
+import { name, version } from '../package.json';
+import {
+  TelemetryClient,
+  setup as AppInsights,
+  defaultClient as AppInsightsClient,
+} from 'applicationinsights';
+import Conf from 'conf';
+
+const config = new Conf();
+const invalidMacAddresses = new Set([
+  '00:00:00:00:00:00',
+  'ff:ff:ff:ff:ff:ff',
+  'ac:de:48:00:11:22',
+]);
+
+// This key is meant to be publically shared. However, I'm adding a simple
+// obfuscation to mitigate key scraping bots on GitHub. The key is split on
+// hypens and base64 encoded without padding.
+// key.split('-').map((x) => x.toString('base64').replace(/=*/, ''))
+const INSTRUMENTATION_KEY = [
```

Comment:
Re hiding this from bots, is there any advantage to using a name that doesn't contain `KEY`?

Reply:
Maybe, but I think this is already overkill. I've seen a number of repositories where they just paste the key in plain text, Microsoft included.